packages/core/test/discovery.test.js

```diff
@@ -748,6 +751,81 @@ describe('Discovery', () => {
     ]));
   });
 
+  it('does not error on cancelled requests', async () => {
+    percy.loglevel('debug');
+
+    let url = 'http://localhost:8000/test';
+    let doc = dedent`
+      <!DOCTYPE html><html><head></head><body><script>
+        async function fetchWithTimeout(resource, options = {}) {
+          const { timeout = 8000 } = options;
+          const controller = new AbortController();
+          const id = setTimeout(() => controller.abort(), timeout);
+          const response = await fetch(resource, {
+            ...options,
+            signal: controller.signal
+          });
+          clearTimeout(id);
+          return response;
+        }
+        setTimeout(() => fetchWithTimeout("${url}", { timeout: 50 }), 10);
+      </script></body></html>
+    `;
+    // above request is cancelled in 50ms, and it will not resolve in 50ms as per following
+    // mock so we can reproduce browser cancelling the request post requesting it
+
+    spyOn(Session.prototype, 'send').and.callFake(function(method, params) {
+      if (method === 'Fetch.continueRequest') {
+        this.log.debug(`Got Fetch.continueRequest ${params.requestId}`);
+        this.log.debug(`Waiting for request to get aborted ${params.requestId}`);
+
+        // waste 1 sec
+        let startTime = new Date().getSeconds();
+        while (startTime === new Date().getSeconds()) {
+          // waste time
+          // We cant use jasmine timer tick here as thats simulated time vs we want to actually
+          // wait for 1 sec
+          // We are waiting for js fetch call in above doc to get cancelled as we have set timeout
+          // of 50ms there
+          // Current real time wait would stall the request for a second so that it cant resolve in
+          // 50 ms. Allowing us to reproduce the race condition
+        }
+        // note, while we wait 1 sec on primary page load request as well, that will not get
+        // aborted so we are good, even if request is delayed
+      }
+      return this.send.and.originalFn.call(this, method, params);
+    });
+
+    server.reply('/', () => [200, 'text/html', doc]);
+
+    server.reply('/test', () => [200, 'text/plain', 'abc']);
+
+    await percy.snapshot({
+      name: 'cancelled request',
+      url: 'http://localhost:8000',
+      enableJavaScript: true
+    });
+
+    await percy.idle();
+
+    expect(logger.stdout).toEqual(jasmine.arrayContaining([
+      '[percy:core] Snapshot taken: cancelled request'
```

Comment:
we're still taking snapshot in this case? or is it that this log line comes first?

Reply:
We will be taking snapshot anyway, this is not an error, it was just falsely shown as one. 
For customers this just avoid error logs in debug logs.